File: code-coaches/medium/snap-crackle-and-pop/snap_crackle_and_pop.cpp
#include <iostream>
#include <vector>

std::string get_sound(int num_rice_krispies)
{
    if (num_rice_krispies % 3 == 0)
        return "Pop";
    else if (num_rice_krispies % 2 != 0)
        return "Snap";
    else
        return "Crackle";
}

int main()
{
    std::vector<int> num_rice_krispies_list(6);
    for (int i = 0; i < 6; i++)
    {
        std::cin >> num_rice_krispies_list[i];
    }

    std::vector<std::string> sounds(6);
    for (int i = 0; i < 6; i++)
    {
        sounds[i] = get_sound(num_rice_krispies_list[i]);
    }

    for (const auto& sound : sounds)
    {
        std::cout << sound << " ";
    }

    return 0;
}

File: code-coaches/medium/snap-crackle-and-pop/SnapCrackleAndPop.java
import java.util.ArrayList;
import java.util.List;
import java.util.Scanner;

public class SnapCrackleAndPop {

    public static String getSound(int numRiceKrispies) {
        if (numRiceKrispies % 3 == 0)
            return "Pop";
        else if (numRiceKrispies % 2 != 0)
            return "Snap";
        else
            return "Crackle";
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        List<Integer> numRiceKrispiesList = new ArrayList<>();
        for (int i = 0; i < 6; i++) {
            int num = scanner.nextInt();
            numRiceKrispiesList.add(num);
        }

        List<String> sounds = new ArrayList<>();
        for (int num : numRiceKrispiesList) {
            String sound = getSound(num);
            sounds.add(sound);
        }

        System.out.println(String.join(" ", sounds));
    }
}
